X509.pm

```diff
@@ -172,6 +173,11 @@ sub Crypt::OpenSSL::X509::subjectaltname {
   die '*** Could not prepare definition: '.$asn->error()
       if !$ok;
 
+  # Microsoft's User Principal Name (Smart Card Logon)
+  my $upn = Convert::ASN1->new or die( "New UPN" );
+  $upn->prepare(q(UPN UTF8String)) or die ( "Prepare UPN" );
+  $asn->registeroid( '1.3.6.1.4.1.311.20.2.3', $upn );
+
```

Comment:
Patch in the issue updated to add support for 1.3.6.1.5.5.7.8.7 ServiceName RFC 4985